src/type-layout.md

```diff
@@ -589,6 +590,22 @@ primitive field will have the ABI of the primitive field.
 Because this representation delegates type layout to another type, it cannot be
 used with any other representation.
 
+## Bit validity of composite types
+
+Each Rust type has "bit validity", which is the set of bit patterns which may
+appear in a value of that type. It is [undefined behavior] to produce a value
```

Comment:
This may be better in its own section about bit validity in general, rather than introducing the concept in its entirety in a section specifically about composite types.